### ISSUE TITLE
LSTM: High loss but accuracy of 1.0 

### ISSUE THREAD

**Issue by DarylWM** [tflearn/tflearn#79]
I have this model for predicting time series values:

```
# Network building
net = tflearn.input_data(shape=[None, days_of_history, 1])
net = tflearn.lstm(net, 300)
net = tflearn.dropout(net, 0.2)
net = tflearn.fully_connected(net, 1, activation='linear')
net = tflearn.regression(net, optimizer='adam', loss='mean_square')

# Training
model = tflearn.DNN(net, clip_gradients=0.)
model.fit(trainX, trainY, n_epoch=70000, validation_set=0.1, show_metric=True)

```
I have 3799 samples of shape (3799, 30, 1), divided into 3419 training samples and 380 validation samples. Each sample is 30 float values in a time series, and I'm trying to predict the next value in the series.

After 1040 epochs, the network has a total loss of 17434.8 and accuracy of 1.0. The starting accuracy in epoch 1 is also 1.0, so I think there's something wrong in my network design. Do you have any suggestions?

**Comment by aymericdamien** [comment]
Accuracy is used for classification, in your case you are doing a regression (as it seems your last layer has only 1 dimension), so that is why accuracy always output 1. You may use R2 metric instead, if your inputs data and targets data has same shape:
```
net = tflearn.regression(net, optimizer='adam', loss='mean_square', metric='R2')
```
I think mean_square is a good choice, it is usually a good common objective function for regressions.
Maybe try to change the optimizer: try SGD and change learning_rate argument:
try: net = tflearn.regression(net, optimizer='sgd', loss='mean_square', learning_rate=0.1)
And change learning rate to 1. or 0.01, 0.001 etc...

**Comment by DarylWM** [comment]
Thanks for the suggestion; I'll give that a try.

To explain a little more what I'm trying to do, in each sample I have a series of 30 values, and I'm trying to predict the next value in the series. So my input shape is (30, 1), and the output shape is (1) representing the next value in the series. Do you think my network structure is correct?

**Comment by DarylWM** [comment]
I changed the regression layer to:
```
sgd = tflearn.optimizers.SGD(learning_rate=0.01, lr_decay=0.96, decay_step=100)
net = tflearn.regression(net, optimizer=sgd, loss='mean_square')
```
then I get loss of 'nan'.
```
Training Step: 24732  | total loss: nan
| SGD | epoch: 458 | loss: nan - acc: 1.0000 | val_loss: nan - val_acc: 1.0000 -- iter: 0000/3419
```
I also tried:
`net = tflearn.regression(net, optimizer='sgd', loss='mean_square', learning_rate=0.1)`
but the loss is always 'nan'.

**Comment by aymericdamien** [comment]
Can you try to lower learning rate? otherwise maybe try 'rmsprop'

**Comment by DarylWM** [comment]
I'm not yet understanding your comment "accuracy is used for classification". I'm training a model to make a prediction, and as training progresses, testing the goodness of the model against the validation set. Shouldn't the accuracy be a measure of the how good the prediction is against the validation targets, and therefore not constantly 1.0?

**Comment by aymericdamien** [comment]
Oh, my meaning is that in the way the accuracy function is made in TFLearn, it can only be used in case of categorical classification (Which means the network return a final vector with every dimension representing a class). The reason it is always 1.0 is because accuracy_op will return the index of the predicted class and compare it to the index of your label data (Or in your case, both vector has only 1 dimension, so it will always "guess right", thus 1.0).

In case of continuous data, you can't use accuracy metric, try "R2" if your data are suitable for it, otherwise you may just use the loss to monitor performance, and compare your training set loss with a validation set loss.

**Comment by DarylWM** [comment]
I see now - thanks.
